Source/WTF/wtf/text/WTFString.h

```diff
@@ -262,15 +262,17 @@ class String final {
     WTF_EXPORT_PRIVATE void convertTo16Bit();
 
     // String::fromUTF8 will return a null string if the input data contains invalid UTF-8 sequences.
-    WTF_EXPORT_PRIVATE static String fromUTF8(std::span<const LChar>);
-    WTF_EXPORT_PRIVATE static String fromUTF8(const LChar*);
-    static String fromUTF8(std::span<const char> characters) { return fromUTF8(std::span { reinterpret_cast<const LChar*>(characters.data()), characters.size() }); }
-    static String fromUTF8(const char* string) { return fromUTF8(reinterpret_cast<const LChar*>(string)); }
-    static String fromUTF8ReplacingInvalidSequences(std::span<const LChar>);
+    WTF_EXPORT_PRIVATE static String fromUTF8(std::span<const char8_t>);
```

Comment:
Given the encoding is unambiguous I think this could just become a constructor, if we don't mind the UTF-8 conversion cost being a little less explicit. Other overloads for other character types would need to continue to follow the named constructor pattern to avoid confusion between Latin-1, ASCII, and UTF-8.